_specifications/specification-3-17.md

```diff
@@ -8173,6 +8188,211 @@ export interface Moniker {
+
+_Request_:
+
+* method: 'typeHierarchy/supertypes'
+* params: `TypeHierarchySupertypesParams` defined as follows:
+
+```typescript
+export interface TypeHierarchySupertypesParams extends
+	WorkDoneProgressParams, PartialResultParams {
+	item: TypeHierarchyItem;
+}
+```
+_Response_:
+
+* result: `TypeHierarchyItem[] | null`
+* partial result: `TypeHierarchyItem[]`
+* error: code and message set in case an exception happens during the 'typeHierarchy/supertypes' request
+
+#### <a href="#typeHierarchy_subtypes" name="typeHierarchy_subtypes" class="anchor">Type Hierarchy Subtypes(:leftwards_arrow_with_hook:)</a>
+
+> *Since version 3.17.0*
+
+The request is sent from the client to the server to resolve the subtypes for a given type hierarchy item. The request doesn't define its own client and server capabilities. It is only issued if a server registers for the [`textDocument/prepareTypeHierarchy` request](#textDocument_prepareTypeHierarchy).
+
+_Request_:
+
+* method: 'typeHierarchy/subtypes'
+* params: `TypeHierarchySubtypesParams` defined as follows:
+
+```typescript
+export interface TypeHierarchySubtypesParams extends
+	WorkDoneProgressParams, PartialResultParams {
+	item: TypeHierarchyItem;
+}
+```
+_Response_:
+
+* result: `TypeHierarchyItem[] | null`
+* partial result: `TypeHierarchyItem[]`
+* error: code and message set in case an exception happens during the 'typeHierarchy/subtypes' request
+
+#### <a href="#typeHierarchy_inheritanceTree" name="typeHierarchy_inheritanceTree" class="anchor">Type Hierarchy Inheritance Tree(:leftwards_arrow_with_hook:)</a>
+
+> *Since version 3.17.0*
+
+The request is sent from the client to the server to resolve the inheritance tree for a given type hierarchy item. The response result of this request is `TreeItem<T>`.
+* `TreeItem<T>` defined as follows:
+```typescript
+export interface TreeItem<T> {
+	data: T;
+	/**
+	 * The children of this TreeItem. When `undefined` the children have not been resolved yet.
+	 */
+	children?: TreeItem<T>[];
+}
+```
+The result represents the inheritance tree related to the specified type hierarchy item. Only single inheritance can be represented this way. The root type is expected as the return value. The request doesn't define its own client and server capabilities. It is only issued if a server has the capability for `TypeHierarchyOptions/inheritanceTreeSuppport`.
+
+_Request_:
+
+* method: 'typeHierarchy/inheritanceTree'
+* params: `TypeHierarchyInheritanceTreeParams` defined as follows:
+
+```typescript
+export interface TypeHierarchyInheritanceTreeParams extends
+	WorkDoneProgressParams {
+	item: TypeHierarchyItem;
+}
+```
+_Response_:
+
+* result: `TreeItem<TypeHierarchyItem> | null`
+* error: code and message set in case an exception happens during the 'typeHierarchy/resolve' request
```

Comment:
Is this supposed to be 'typeHierarchy/inheritanceTree'?

Reply:
Yes. I'll fix it in the next commit.